File: src/python/18_operacje_na_plikach/zad12.py
"""
Otrzymujesz dwa napisy reprezentujace sciezki folderow. Przenies wszystkie pliki 
csv z pierwszego folderu (oraz wszystkich jego podfolderow) do drugiego folderu.
"""

import pathlib


def przenies_pliki_csv(folder_1, folder_2):
    """
    Funkcja przenosi pliki csv z folderu folder_1 do folderu folder_2.
    """
    for plik in pathlib.Path(folder_1).glob("**/*.csv"):
        plik.rename(pathlib.Path(folder_2) / plik.name)


def test_przenies_pliki_csv():

    # stworz foldery testowe
    folder_1 = pathlib.Path("test_1")
    folder_1.mkdir(parents=True, exist_ok=True)
    folder_2 = pathlib.Path("test_2")
    folder_2.mkdir(parents=True, exist_ok=True)

    # stworz pliki testowe
    plik_1 = folder_1 / "plik_1.csv"
    plik_1.touch()
    plik_2 = folder_1 / "plik_2.csv"
    plik_2.touch()

    # przenies pliki
    przenies_pliki_csv(folder_1, folder_2)

    # sprawdz czy pliki zostaly przeniesione
    assert pathlib.Path(folder_2 / "plik_1.csv").exists()
    assert pathlib.Path(folder_2 / "plik_2.csv").exists()
    assert not pathlib.Path(folder_1 / "plik_1.csv").exists()
    assert not pathlib.Path(folder_1 / "plik_2.csv").exists()

    # usun foldery testowe
    import shutil

    shutil.rmtree("test_1")


if __name__ == "__main__":

    test_przenies_pliki_csv()

File: src/js/18_operacje_na_plikach/zad12.js
/*
Otrzymujesz dwa napisy reprezentujace sciezki folderow. Przenies wszystkie pliki
csv z pierwszego folderu (oraz wszystkich jego podfolderow) do drugiego folderu.
*/

const fs = require("fs").promises;
const path = require("path");

przeniesPliki = async (zrodlo, cel, rozszerzenie) => {
  const dane = await fs.readdir(zrodlo);
  const pliki = dane.filter((plik) => plik.endsWith(rozszerzenie));
  for (const plik of pliki) {
    const sciezkaPliku = path.join(zrodlo, plik);
    const sciezkaDocelowa = path.join(cel, plik);
    await fs.copyFile(sciezkaPliku, sciezkaDocelowa);
    await fs.unlink(sciezkaPliku);
  }
};

test1 = async () => {
  const folder1 = "temp_dir1";
  const folder2 = "temp_dir2";

  await fs.mkdir(folder1);
  await fs.mkdir(folder2);

  const sciezkaPliku1 = path.join(folder1, "plik1.csv");
  const sciezkaPliku2 = path.join(folder1, "plik2.csv");

  sciezki = [sciezkaPliku1, sciezkaPliku2];

  for (const sciezkaPliku of sciezki) {
    await fs.writeFile(sciezkaPliku, "test1; test2");
  }

  await przeniesPliki(folder1, folder2, ".csv");

  const nowaSciezkaPliku1 = path.join(folder2, "plik1.csv");
  const nowaSciezkaPliku2 = path.join(folder2, "plik2.csv");

  const noweSciezki = [nowaSciezkaPliku1, nowaSciezkaPliku2];

  for (const nowaSciezkaPliku of noweSciezki) {
    const stat = await fs.lstat(nowaSciezkaPliku);
    if (!stat.isFile()) {
      throw new Error(
        `Assertion error line 32: ${nowaSciezkaPliku} does not exist`
      );
    }
  }

  for (const sciezkaPliku of sciezki) {
    try {
      await fs.access(sciezkaPliku);
      throw new Error(`Assertion error line 38: ${sciezkaPliku} exists`);
    } catch (err) {
      if (err.code !== "ENOENT") {
        throw err;
      }
    }
  }

  await fs.rmdir(folder1, {
    recursive: true,
  });
  await fs.rmdir(folder2, {
    recursive: true,
  });
};

main = async () => {
  await test1();
};

main();
